app/lib/features/profile/pages/my_profile_page.dart

```diff
@@ -60,25 +70,27 @@ class _ChangeDisplayNameState extends State<ChangeDisplayName> {
           child: const Text('Cancel'),
         ),
         TextButton(
-          onPressed: () {
-            if (_formKey.currentState!.validate()) {
-              final currentUserName = account.profile.displayName;
-              final newDisplayName = newUsername.text;
-              if (currentUserName != newDisplayName) {
-                Navigator.pop(context, newDisplayName);
-              } else {
-                Navigator.pop(context, null);
-              }
-              return;
-            }
-          },
+          onPressed: () => onSubmit(context),
           child: const Text('Submit'),
         ),
       ],
     );
   }
+
+  void onSubmit(BuildContext context) {
+    if (!_formKey.currentState!.validate()) {
+      return;
```

Comment:
Could use some informational dialog or snackbar showing fields aren't valid rather than just leave it empty?

Reply:
Fixed by https://github.com/acterglobal/a3/pull/926/commits/bbe0c9f5f45011716730fc3c8c683c0e92309883